### PR TITLE
feat: add serviceaccount

### LINKED ISSUE #70: [zabbix-community/zabbix] Add option to use custom service account
**Is your feature request related to a problem? Please describe.**
We're trying to setup this chart on an AKS cluster using the secrets-store CSI. For this we need to be able to add labels to a SA. 
This chart doesn't have that functionality so we tried to create our own SA, but we cannot specify this in the chart values.

**Describe the solution you'd like**
A way to provide a custom SA

**Describe alternatives you've considered**
We need to add labels to the SA. This could also be implemented instead of a custom SA, although it probably would be useful to have both.

**Additional context**
-

### PR DESCRIPTION
**Pull request by syphernl**
#### What this PR does / why we need it:
Adds a configurable ServiceAccount to the chart.

#### Which issue this PR fixes
  - fixes #70 

#### Checklist

- [x] [DCO](https://github.com/zabbix-community/helm-zabbix/blob/master/CONTRIBUTING.md) signed
- [ ] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: charts/zabbix/values.yaml [hunk 1]
```diff
@@ -631,25 +643,41 @@ karpenter:
...
+    values: ["spot", "on-demand"]
   # -- Optional, configures IMDS for the instance
   metadataOptions:
     httpEndpoint: enabled
     httpProtocolIPv6: disabled
     httpPutResponseHopLimit: 2
     httpTokens: required
 
+serviceAccount:
+  # Specifies whether a service account should be created.
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:669]: Hi @syphernl1

Can you change this line?

`# -- Specifies whether a service account should be created.`

File: charts/zabbix/values.yaml [hunk 2]
```diff
@@ -631,25 +643,41 @@ karpenter:
...
+  create: true
+
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template.
+  name: ""
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:674]: Hi @syphernl1

Can you change these lines?

```
  # -- The name of the service account to use. If not set and create is true, a name is generated using the fullname template.
  name: ""
```

File: charts/zabbix/values.yaml [hunk 3]
```diff
@@ -631,25 +643,41 @@ karpenter:
...
+
+  # Optional additional annotations to add to the Service Account.
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:676]: Hi @syphernl1

Can you change this line?

`  # -- Optional additional annotations to add to the Service Account.`

File: charts/zabbix/values.yaml [hunk 4]
```diff
@@ -631,25 +643,41 @@ karpenter:
...
+  annotations: {}
+
+  # Optional additional labels to add to the Service Account.
+  labels: {}
+
+  # Automount API credentials for a Service Account.
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:682]: Hi @syphernl1

Can you change this line?

`# -- Automount API credentials for a Service Account.`

**Comment by syphernl** [comment]
@aeciopires Changes applied.

**Comment by syphernl** [comment]
For some reason I am still seeing that the ha cleanup job uses the `default` SA rather than the configured one, this needs to be looked into.

**Review by aeciopires — CHANGES_REQUESTED** [review]
Hello @syphernl 

Thank you very much for your contribution!

I missed adding the ServiceAccount for PostgreSQL and Zabbix Proxy as well. Could you do this?

Another thing... I think that when I merge your other PR first or this one, many conflicts will appear because they are both messing with the contents of the same files.

I think it would be good for you to unify the 2 PRs? What do you think?

**Review by aeciopires — APPROVED** [review]

**Comment by aeciopires** [comment]
Hi @syphernl 

I merged a PR that was pending, but I haven't published a version yet... I'll wait for your changes...

Could you synchronize your 2 PRs with the master branch to get my changes and correct any conflicts?

**Comment by syphernl** [comment]
Done

**Pull request merged**